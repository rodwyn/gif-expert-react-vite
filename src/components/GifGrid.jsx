import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {
  const {images, isLoading} = useFetchGifs(category);

  return (
    <>
        <h1>{category}</h1>
        {
          isLoading && <h2>Cargando...</h2>
        }
        
        <div className='card-grid'>
          {
            images.map((image) => (
              <GifItem key={image.id} {...image}  />
            ))
          }
        </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};
