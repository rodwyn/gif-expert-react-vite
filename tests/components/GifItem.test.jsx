import { render,screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Testing <GifItem /> component.', () => {
    const title = 'pikachu';
    const url = 'https://pokemon.com/pikachu.jpg';

    test('should match with snapshot', () => { 
        const { container } =  render(<GifItem title={title} url={url} />);
        
        expect(container).toMatchSnapshot();
     });

     test('should render url and alt text correctly.', () => { 
        render(<GifItem title={title} url={url} />);
        const { src, alt} = screen.getByRole('img'); 

        expect(src).toBe(url);
        expect(alt).toBe(title);
      });

      test('should render title correctly.', () => { 
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
       })
 });
