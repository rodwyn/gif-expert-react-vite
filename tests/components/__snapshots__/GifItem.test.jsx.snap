// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem /> component. should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="pikachu"
      src="https://pokemon.com/pikachu.jpg"
    />
    <p>
      pikachu
    </p>
  </div>
</div>
`;
